youtube_dl/extractor/zingmp3.py

```diff
@@ -1,23 +1,52 @@
 # coding: utf-8
 from __future__ import unicode_literals
 
+import datetime
+import hashlib
+import hmac
+import re
+from urllib.parse import urlencode
+
 from .common import InfoExtractor
 from ..utils import (
-    ExtractorError,
     int_or_none,
+    try_get,
 )
 
 
 class ZingMp3BaseIE(InfoExtractor):
-    _VALID_URL_TMPL = r'https?://(?:mp3\.zing|zingmp3)\.vn/(?:%s)/[^/]+/(?P<id>\w+)\.html'
+    _VALID_URL_TMPL = r'https?://(?:mp3\.zing|zingmp3)\.vn/(?P<type>(?:%s))/[^/]+/(?P<id>\w+)(?:\.html|\?)'
```

Comment:
As the `<id>` match will end on `[.#?]`, do you need the `.html` match at all?

Reply:
Updated, ths you.